packages/react-native-sdk/src/contexts/internal/SnapshotContext.tsx

```diff
@@ -0,0 +1,152 @@
+import {
+  StreamVideoParticipant,
+  type VideoTrackType,
+  getLogger,
+} from '@stream-io/video-client';
+import React, {
+  createContext,
+  useContext,
+  useRef,
+  RefObject,
+  useCallback,
+  useMemo,
+} from 'react';
+import { NativeModules, findNodeHandle, Platform } from 'react-native';
+
+const { StreamVideoReactNative } = NativeModules;
+
+type SnapshotContextType = {
+  register: (
+    participant: StreamVideoParticipant,
+    videoTrackType: VideoTrackType,
+    ref: RefObject<any>,
+  ) => void;
+  deregister: (
+    participant: StreamVideoParticipant,
+    videoTrackType: VideoTrackType,
+  ) => void;
+  take: (
+    participant: StreamVideoParticipant,
+    videoTrackType: VideoTrackType,
+  ) => Promise<string | null>;
+};
+
+// Create the context with a default undefined value
+const SnapshotContext = createContext<SnapshotContextType | undefined>(
+  undefined,
+);
+
+// Reference map type
+type RefMap = Map<string, RefObject<any>>;
+
+export const SnapshotProvider = ({ children }: React.PropsWithChildren<{}>) => {
+  // Use a ref to store the map of participant IDs to their view refs
+  const participantRefs = useRef<RefMap>(new Map());
```

Comment:
This creates a `new Map()` on every render and puts pressure on the GC.
Do we have to use a ref here?

Reply:
unnecessary and its now removed